lisa/sut_orchestrator/azure/features.py

```diff
@@ -783,3 +790,126 @@ def get_azure_disk_type(disk_type: schema.DiskType) -> str:
     schema.DiskType.StandardHDDLRS: "Standard_LRS",
     schema.DiskType.StandardSSDLRS: "StandardSSD_LRS",
 }
+
+
+class Resize(AzureFeatureMixin, features.Resize):
+    def resize(self, resize_action: ResizeAction = None) -> None:
+        platform: AzurePlatform = self._platform
+        compute_client = get_compute_client(platform)
+        vm_operations = compute_client.virtual_machines
+        node_context = get_node_context(self._node)
+        node_runbook = self._node.capability.get_extended_runbook(AzureNodeSchema)
+
+        # get compatible vm sizes.
+        available_sizes = vm_operations.list_available_sizes(
+            node_context.resource_group_name, node_context.vm_name
+        )
+        vm_sizes_with_capability = platform.get_eligible_vm_sizes(
+            node_runbook.location, self._log
+        )  # add location in node_context
+
+        current_vm_size = next(
+            (x for x in vm_sizes_with_capability if x.vm_size == node_runbook.vm_size),
+            None,
+        )
+        assert current_vm_size, "cannot find current vm size in eligible list"
+
+        resizable_vmsizes: List[AzureCapability] = []
+        for size in available_sizes:
```

Comment:
add comments above each loop to explain its purpose.

Reply:
done